services/movie_session.py

```diff
@@ -0,0 +1,62 @@
+from db.models import MovieSession
+from db.models import CinemaHall
+from db.models import Movie
+
+
+def create_movie_session(
+        movie_show_time: str,
+        movie_id: int,
+        cinema_hall_id: int
+) -> None:
+    MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie=Movie.objects.get(id=movie_id),
+        cinema_hall=CinemaHall.objects.get(id=cinema_hall_id),
+    )
+
+
+def get_movies_sessions(
+        session_date: str = None
+) -> MovieSession.objects:
+    queryset = MovieSession.objects.all()
+
+    if session_date is not None:
+        queryset = MovieSession.objects.filter(
```

Comment:
This function features unnecessary database calls. It can be improved in one of the following ways:
```python
if condition:
    return Model.objects.filter(...)
return Model.objects.all()
```
```python
qs = Model.objects.all()
if condition:
    qs = qs.filter(...)
return qs
```

Reply:
what is the difference

Reply:
Oh, wait. Your implementation actually doesn't call database twice.
Either way, it's a bad practice to put something like a queryset into a variable and not reuse it.

Reply:
The difference is that in your solution, the function creates a queryset and immediately throws it away if `session_date` is not `None`. There's no reason to do it like that. You should either continue to use this queryset or not create it at all.